src/devices/cpu/z180/z180.cpp

```diff
@@ -1936,17 +1751,41 @@ void z180_device::device_reset()
 	z180_mmu();
 }
 
+void z180_device::device_add_mconfig(machine_config &config)
+{
+	Z180ASCI_CHANNEL_0(config, m_asci[0], DERIVED_CLOCK(1,2));
+	m_asci[0]->txa_handler().set(FUNC(z180_device::txa0_trampoline_w));
+	m_asci[0]->rts_handler().set(FUNC(z180_device::rts0_trampoline_w));
+
+	Z180ASCI_CHANNEL_1(config, m_asci[1], DERIVED_CLOCK(1,2));
+	m_asci[1]->txa_handler().set(FUNC(z180_device::txa1_trampoline_w));
+}
+
+void z8s180_device::device_add_mconfig(machine_config &config)
+{
+	Z180ASCI_EXT_CHANNEL_0(config, m_asci[0], DERIVED_CLOCK(1,2));
+	m_asci[0]->txa_handler().set(FUNC(z8s180_device::txa0_trampoline_w));
+	m_asci[0]->rts_handler().set(FUNC(z8s180_device::rts0_trampoline_w));
+
+	Z180ASCI_EXT_CHANNEL_1(config, m_asci[1], DERIVED_CLOCK(1,2));
+	m_asci[1]->txa_handler().set(FUNC(z8s180_device::txa1_trampoline_w));
+}
+
 void z8s180_device::device_reset()
 {
 	z180_device::device_reset();
 
-	m_asci_ext[0] = 0x00;
-	m_asci_ext[1] = 0x00;
 	m_cmr = 0x00;
 	m_ccr = 0x00;
 	notify_clock_changed();
 }
 
+void z8s180_device::device_clock_changed()
+{
+	m_asci[0]->set_clock((m_cmr & 0x80) ? (clock() * 2) : (m_ccr & 0x80) ? clock() : (clock() / 2));
+	m_asci[1]->set_clock((m_cmr & 0x80) ? (clock() * 2) : (m_ccr & 0x80) ? clock() : (clock() / 2));
+}
+
```

Comment:
You need to use `DERIVED_CLOCK` here – if you don’t, it will cause problems when the CPU itself is configured with `DERIVED_CLOCK` and its input clock is changed.